Shepherd: judge validation failure skips doctor loop entirely

## Summary

When the shepherd's judge phase validation fails (neither `loom:pr` nor `loom:changes-requested` label is present), the shepherd exits with an error instead of entering the Judge→Doctor retry loop. The doctor loop requires `changes_requested` to be detected in the judge result, but when the judge produces *no outcome at all*, the loop never triggers.

This means any judge failure — timeout, worker crash, API error — becomes a terminal shepherd failure rather than a recoverable state.

## Observed Behavior

```
[WARN] [force-mode] Label validation failed for PR #106, attempting fallback approval detection
[INFO] [force-mode] No approval comment found in PR — fallback denied
[ERROR] judge phase validation failed
```

The shepherd exits immediately. No doctor is invoked. No retry of the judge happens.

## Expected Behavior

The Judge→Doctor loop (cli.py lines 315-400) should handle three judge outcomes:

1. **Approved** (`loom:pr`) → proceed to merge
2. **Changes requested** (`loom:changes-requested`) → invoke doctor, re-run judge
3. **No outcome / validation failure** → this is the gap

For case 3, the shepherd should:
- Retry the judge phase (at least once) before giving up
- Or treat a failed judge as "changes requested" and enter the doctor loop in `--merge` mode
- Or mark the issue as `loom:blocked` with diagnostics rather than silently failing

## Impact

In `--merge` mode, users expect full automation. A silent judge failure with no recovery path defeats the purpose of the merge flag. The user must manually diagnose and re-run, which is the friction the shepherd is supposed to eliminate.

## Related

This is closely related to the judge worker silently failing (see companion issue), but this issue is about the shepherd's control flow — even if the judge worker bug is fixed, the shepherd should handle unexpected validation failures gracefully.

--- a/loom-tools/src/loom_tools/shepherd/cli.py
+++ b/loom-tools/src/loom_tools/shepherd/cli.py
@@ -415,6 +415,7 @@ def orchestrate(ctx: ShepherdContext) -> int:
 
         # ─── PHASE 4/5: Judge/Doctor Loop ─────────────────────────────────
         doctor_attempts = 0
+        judge_retries = 0
         pr_approved = False
 
         # Check for --from merge skip
@@ -441,8 +442,34 @@ def orchestrate(ctx: ShepherdContext) -> int:
                 raise ShutdownSignal(result.message)
 
             if result.status in (PhaseStatus.FAILED, PhaseStatus.STUCK):
-                log_error(result.message)
+                # Judge returned FAILED/STUCK with no label outcome.
+                # Retry the judge phase before giving up (defense-in-depth
+                # for cases where the judge worker silently fails without
+                # submitting a review, leaving no loom:pr or
+                # loom:changes-requested label).
+                if judge_retries < ctx.config.judge_max_retries:
+                    judge_retries += 1
+                    log_warning(
+                        f"Judge phase failed ({result.message}), "
+                        f"retrying ({judge_retries}/{ctx.config.judge_max_retries})"
+                    )
+                    ctx.report_milestone(
+                        "judge_retry",
+                        attempt=judge_retries,
+                        max_retries=ctx.config.judge_max_retries,
+                        reason=result.message,
+                    )
+                    continue
+
+                log_error(
+                    f"Judge phase failed after {judge_retries} "
+                    f"retry attempt(s): {result.message}"
+                )
+                _mark_judge_exhausted(ctx, judge_retries)
                 return 1
+
+            # Judge succeeded — reset retry counter for this loop iteration
+            judge_retries = 0
 
             if result.data.get("approved"):
                 pr_approved = True
@@ -496,7 +523,27 @@ def orchestrate(ctx: ShepherdContext) -> int:
                     status="success",
                 )
             else:
-                log_error(result.message)
+                # Unexpected result — neither approved, changes_requested,
+                # nor FAILED/STUCK. Treat as a judge failure and retry.
+                if judge_retries < ctx.config.judge_max_retries:
+                    judge_retries += 1
+                    log_warning(
+                        f"Judge returned unexpected result ({result.message}), "
+                        f"retrying ({judge_retries}/{ctx.config.judge_max_retries})"
+                    )
+                    ctx.report_milestone(
+                        "judge_retry",
+                        attempt=judge_retries,
+                        max_retries=ctx.config.judge_max_retries,
+                        reason=result.message,
+                    )
+                    continue
+
+                log_error(
+                    f"Judge phase returned unexpected result after {judge_retries} "
+                    f"retry attempt(s): {result.message}"
+                )
+                _mark_judge_exhausted(ctx, judge_retries)
                 return 1
 
         # Print skipped header if Doctor never ran (Judge approved first try)
@@ -690,6 +737,60 @@ def _mark_doctor_exhausted(ctx: ShepherdContext) -> None:
             str(ctx.config.issue),
             "--body",
             f"**Shepherd blocked**: Doctor could not resolve Judge feedback after {ctx.config.doctor_max_retries} attempts.",
+        ],
+        cwd=ctx.repo_root,
+        capture_output=True,
+        check=False,
+    )
+
+
+def _mark_judge_exhausted(ctx: ShepherdContext, retries: int) -> None:
+    """Mark issue as blocked due to judge retry exhaustion."""
+    import subprocess
+
+    # Atomic transition: loom:building -> loom:blocked
+    subprocess.run(
+        [
+            "gh",
+            "issue",
+            "edit",
+            str(ctx.config.issue),
+            "--remove-label",
+            "loom:building",
+            "--add-label",
+            "loom:blocked",
+        ],
+        cwd=ctx.repo_root,
+        capture_output=True,
+        check=False,
+    )
+
+    # Record blocked reason and update systematic failure tracking
+    from loom_tools.common.systematic_failure import (
+        detect_systematic_failure,
+        record_blocked_reason,
+    )
+
+    record_blocked_reason(
+        ctx.repo_root,
+        ctx.config.issue,
+        error_class="judge_exhausted",
+        phase="judge",
+        details=f"judge failed after {retries} retry attempt(s)",
+    )
+    detect_systematic_failure(ctx.repo_root)
+
+    # Add comment
+    subprocess.run(
+        [
+            "gh",
+            "issue",
+            "comment",
+            str(ctx.config.issue),
+            "--body",
+            f"**Shepherd blocked**: Judge phase failed after {retries} retry attempt(s). "
+            "No approval or changes-requested outcome was produced. "
+            "This may indicate a judge worker failure — see #1908 for related diagnostics.",
         ],
         cwd=ctx.repo_root,
         capture_output=True,

--- a/loom-tools/src/loom_tools/shepherd/config.py
+++ b/loom-tools/src/loom_tools/shepherd/config.py
@@ -80,6 +80,9 @@ class ShepherdConfig:
     doctor_max_retries: int = field(
         default_factory=lambda: env_int("LOOM_DOCTOR_MAX_RETRIES", 3)
     )
+    judge_max_retries: int = field(
+        default_factory=lambda: env_int("LOOM_JUDGE_MAX_RETRIES", 1)
+    )
     stuck_max_retries: int = field(
         default_factory=lambda: env_int("LOOM_STUCK_MAX_RETRIES", 1)
     )

--- a/loom-tools/tests/shepherd/test_cli.py
+++ b/loom-tools/tests/shepherd/test_cli.py
@@ -9,7 +9,13 @@ from unittest.mock import MagicMock, call, patch
 
 import pytest
 
-from loom_tools.shepherd.cli import _create_config, _parse_args, main, orchestrate
+from loom_tools.shepherd.cli import (
+    _create_config,
+    _mark_judge_exhausted,
+    _parse_args,
+    main,
+    orchestrate,
+)
 from loom_tools.shepherd.config import ExecutionMode, Phase, ShepherdConfig
 from loom_tools.shepherd.phases import PhaseResult, PhaseStatus
 
@@ -745,3 +751,376 @@ class TestApprovalPhaseSummary:
         )
         new_output = f"Approval ({result.data.get('summary', result.message)})"
         assert new_output == "Approval (human approved)"  # correct
+
+
+def _failed_result(phase: str = "", message: str = "phase failed") -> PhaseResult:
+    """Create a failed PhaseResult."""
+    return PhaseResult(status=PhaseStatus.FAILED, message=message, phase_name=phase)
+
+
+class TestJudgeRetry:
+    """Test judge retry logic when judge phase fails (issue #1909)."""
+
+    @patch("loom_tools.shepherd.cli.MergePhase")
+    @patch("loom_tools.shepherd.cli.JudgePhase")
+    @patch("loom_tools.shepherd.cli.BuilderPhase")
+    @patch("loom_tools.shepherd.cli.ApprovalPhase")
+    @patch("loom_tools.shepherd.cli.CuratorPhase")
+    @patch("loom_tools.shepherd.cli.time")
+    def test_judge_failure_triggers_retry_then_succeeds(
+        self,
+        mock_time: MagicMock,
+        MockCurator: MagicMock,
+        MockApproval: MagicMock,
+        MockBuilder: MagicMock,
+        MockJudge: MagicMock,
+        MockMerge: MagicMock,
+    ) -> None:
+        """Judge FAILED on first call, SUCCESS with approved on second — should complete."""
+        mock_time.time = MagicMock(side_effect=[
+            0,     # start_time
+            0,     # approval phase_start
+            5,     # approval elapsed
+            # Judge attempt 1 (fails)
+            5,     # judge phase_start
+            10,    # judge elapsed
+            # Judge attempt 2 (succeeds with approved)
+            10,    # judge phase_start
+            20,    # judge elapsed
+            # Merge
+            20,    # merge phase_start
+            25,    # merge elapsed
+            25,    # duration calc
+        ])
+
+        ctx = _make_ctx(start_from=Phase.BUILDER)
+        # Default judge_max_retries=1, so 1 retry allowed
+        assert ctx.config.judge_max_retries == 1
+
+        curator_inst = MockCurator.return_value
+        curator_inst.should_skip.return_value = (True, "skipped via --from")
+        MockApproval.return_value.run.return_value = _success_result("approval")
+        builder_inst = MockBuilder.return_value
+        builder_inst.should_skip.return_value = (True, "skipped via --from")
+
+        # Judge: first call fails, second approves
+        judge_inst = MockJudge.return_value
+        judge_inst.should_skip.return_value = (False, "")
+        judge_inst.run.side_effect = [
+            _failed_result("judge", "judge phase validation failed"),
+            _success_result("judge", approved=True),
+        ]
+
+        MockMerge.return_value.run.return_value = _success_result("merge", merged=True)
+
+        result = orchestrate(ctx)
+        assert result == 0
+
+        # Verify judge_retry milestone was reported
+        retry_calls = [
+            c for c in ctx.report_milestone.call_args_list
+            if c[0][0] == "judge_retry"
+        ]
+        assert len(retry_calls) == 1
+        assert retry_calls[0].kwargs["attempt"] == 1
+
+    @patch("loom_tools.shepherd.cli._mark_judge_exhausted")
+    @patch("loom_tools.shepherd.cli.JudgePhase")
+    @patch("loom_tools.shepherd.cli.BuilderPhase")
+    @patch("loom_tools.shepherd.cli.ApprovalPhase")
+    @patch("loom_tools.shepherd.cli.CuratorPhase")
+    @patch("loom_tools.shepherd.cli.time")
+    def test_judge_retry_exhaustion_marks_blocked(
+        self,
+        mock_time: MagicMock,
+        MockCurator: MagicMock,
+        MockApproval: MagicMock,
+        MockBuilder: MagicMock,
+        MockJudge: MagicMock,
+        mock_mark_exhausted: MagicMock,
+    ) -> None:
+        """Judge always fails — should call _mark_judge_exhausted and return 1."""
+        mock_time.time = MagicMock(side_effect=[
+            0,     # start_time
+            0,     # approval phase_start
+            5,     # approval elapsed
+            # Judge attempt 1 (fails — triggers retry)
+            5,     # judge phase_start
+            10,    # judge elapsed
+            # Judge attempt 2 (fails again — retries exhausted)
+            10,    # judge phase_start
+            15,    # judge elapsed
+        ])
+
+        ctx = _make_ctx(start_from=Phase.BUILDER)
+
+        curator_inst = MockCurator.return_value
+        curator_inst.should_skip.return_value = (True, "skipped via --from")
+        MockApproval.return_value.run.return_value = _success_result("approval")
+        builder_inst = MockBuilder.return_value
+        builder_inst.should_skip.return_value = (True, "skipped via --from")
+
+        # Judge: always fails
+        judge_inst = MockJudge.return_value
+        judge_inst.should_skip.return_value = (False, "")
+        judge_inst.run.return_value = _failed_result("judge", "validation failed")
+
+        result = orchestrate(ctx)
+        assert result == 1
+
+        # Verify _mark_judge_exhausted was called with retry count (1):
+        # first fail triggers retry (judge_retries=1), second fail exhausts retries
+        mock_mark_exhausted.assert_called_once_with(ctx, 1)
+
+    @patch("loom_tools.shepherd.cli.MergePhase")
+    @patch("loom_tools.shepherd.cli.JudgePhase")
+    @patch("loom_tools.shepherd.cli.BuilderPhase")
+    @patch("loom_tools.shepherd.cli.ApprovalPhase")
+    @patch("loom_tools.shepherd.cli.CuratorPhase")
+    @patch("loom_tools.shepherd.cli.time")
+    def test_approved_path_unchanged_with_retry_logic(
+        self,
+        mock_time: MagicMock,
+        MockCurator: MagicMock,
+        MockApproval: MagicMock,
+        MockBuilder: MagicMock,
+        MockJudge: MagicMock,
+        MockMerge: MagicMock,
+    ) -> None:
+        """Approved outcome on first try should work identically to before."""
+        mock_time.time = MagicMock(side_effect=[
+            0,     # start_time
+            0,     # approval phase_start
+            5,     # approval elapsed
+            5,     # judge phase_start
+            15,    # judge elapsed
+            15,    # merge phase_start
+            20,    # merge elapsed
+            20,    # duration calc
+        ])
+
+        ctx = _make_ctx(start_from=Phase.BUILDER)
+
+        curator_inst = MockCurator.return_value
+        curator_inst.should_skip.return_value = (True, "skipped via --from")
+        MockApproval.return_value.run.return_value = _success_result("approval")
+        builder_inst = MockBuilder.return_value
+        builder_inst.should_skip.return_value = (True, "skipped via --from")
+
+        judge_inst = MockJudge.return_value
+        judge_inst.should_skip.return_value = (False, "")
+        judge_inst.run.return_value = _success_result("judge", approved=True)
+
+        MockMerge.return_value.run.return_value = _success_result("merge", merged=True)
+
+        result = orchestrate(ctx)
+        assert result == 0
+
+        # No retry milestones should be reported
+        retry_calls = [
+            c for c in ctx.report_milestone.call_args_list
+            if c[0][0] == "judge_retry"
+        ]
+        assert len(retry_calls) == 0
+
+    @patch("loom_tools.shepherd.cli.MergePhase")
+    @patch("loom_tools.shepherd.cli.DoctorPhase")
+    @patch("loom_tools.shepherd.cli.JudgePhase")
+    @patch("loom_tools.shepherd.cli.BuilderPhase")
+    @patch("loom_tools.shepherd.cli.ApprovalPhase")
+    @patch("loom_tools.shepherd.cli.CuratorPhase")
+    @patch("loom_tools.shepherd.cli.time")
+    def test_changes_requested_path_unchanged_with_retry_logic(
+        self,
+        mock_time: MagicMock,
+        MockCurator: MagicMock,
+        MockApproval: MagicMock,
+        MockBuilder: MagicMock,
+        MockJudge: MagicMock,
+        MockDoctor: MagicMock,
+        MockMerge: MagicMock,
+    ) -> None:
+        """Changes-requested -> doctor -> approved flow should work as before."""
+        mock_time.time = MagicMock(side_effect=[
+            0,     # start_time
+            0,     # approval phase_start
+            5,     # approval elapsed
+            # Judge 1: changes requested
+            5,     # judge phase_start
+            15,    # judge elapsed
+            # Doctor
+            15,    # doctor phase_start
+            25,    # doctor elapsed
+            # Judge 2: approved
+            25,    # judge phase_start
+            35,    # judge elapsed
+            # Merge
+            35,    # merge phase_start
+            40,    # merge elapsed
+            40,    # duration calc
+        ])
+
+        ctx = _make_ctx(start_from=Phase.BUILDER)
+
+        curator_inst = MockCurator.return_value
+        curator_inst.should_skip.return_value = (True, "skipped via --from")
+        MockApproval.return_value.run.return_value = _success_result("approval")
+        builder_inst = MockBuilder.return_value
+        builder_inst.should_skip.return_value = (True, "skipped via --from")
+
+        judge_inst = MockJudge.return_value
+        judge_inst.should_skip.return_value = (False, "")
+        judge_inst.run.side_effect = [
+            _success_result("judge", changes_requested=True),
+            _success_result("judge", approved=True),
+        ]
+
+        MockDoctor.return_value.run.return_value = _success_result("doctor")
+        MockMerge.return_value.run.return_value = _success_result("merge", merged=True)
+
+        result = orchestrate(ctx)
+        assert result == 0
+
+        # No judge retry milestones
+        retry_calls = [
+            c for c in ctx.report_milestone.call_args_list
+            if c[0][0] == "judge_retry"
+        ]
+        assert len(retry_calls) == 0
+
+        # Doctor milestone should be present
+        doctor_calls = [
+            c for c in ctx.report_milestone.call_args_list
+            if c[0][0] == "phase_completed" and c.kwargs.get("phase") == "doctor"
+        ]
+        assert len(doctor_calls) == 1
+
+    @patch("loom_tools.shepherd.cli.MergePhase")
+    @patch("loom_tools.shepherd.cli.JudgePhase")
+    @patch("loom_tools.shepherd.cli.BuilderPhase")
+    @patch("loom_tools.shepherd.cli.ApprovalPhase")
+    @patch("loom_tools.shepherd.cli.CuratorPhase")
+    @patch("loom_tools.shepherd.cli.time")
+    def test_judge_failure_retry_reports_milestones(
+        self,
+        mock_time: MagicMock,
+        MockCurator: MagicMock,
+        MockApproval: MagicMock,
+        MockBuilder: MagicMock,
+        MockJudge: MagicMock,
+        MockMerge: MagicMock,
+    ) -> None:
+        """Judge retry milestone should include attempt count and reason."""
+        mock_time.time = MagicMock(side_effect=[
+            0,     # start_time
+            0,     # approval phase_start
+            5,     # approval elapsed
+            # Judge attempt 1 (fails)
+            5,     # judge phase_start
+            10,    # judge elapsed
+            # Judge attempt 2 (approves)
+            10,    # judge phase_start
+            20,    # judge elapsed
+            # Merge
+            20,    # merge phase_start
+            25,    # merge elapsed
+            25,    # duration calc
+        ])
+
+        ctx = _make_ctx(start_from=Phase.BUILDER)
+
+        curator_inst = MockCurator.return_value
+        curator_inst.should_skip.return_value = (True, "skipped via --from")
+        MockApproval.return_value.run.return_value = _success_result("approval")
+        builder_inst = MockBuilder.return_value
+        builder_inst.should_skip.return_value = (True, "skipped via --from")
+
+        judge_inst = MockJudge.return_value
+        judge_inst.should_skip.return_value = (False, "")
+        judge_inst.run.side_effect = [
+            _failed_result("judge", "validation failed"),
+            _success_result("judge", approved=True),
+        ]
+
+        MockMerge.return_value.run.return_value = _success_result("merge", merged=True)
+
+        result = orchestrate(ctx)
+        assert result == 0
+
+        # Verify milestone content
+        retry_calls = [
+            c for c in ctx.report_milestone.call_args_list
+            if c[0][0] == "judge_retry"
+        ]
+        assert len(retry_calls) == 1
+        assert retry_calls[0].kwargs["attempt"] == 1
+        assert retry_calls[0].kwargs["max_retries"] == 1
+        assert "validation failed" in retry_calls[0].kwargs["reason"]
+
+
+class TestMarkJudgeExhausted:
+    """Test _mark_judge_exhausted helper function."""
+
+    @patch("subprocess.run")
+    def test_transitions_labels(self, mock_run: MagicMock) -> None:
+        """Should run gh command to transition loom:building -> loom:blocked."""
+        mock_run.return_value = MagicMock(returncode=0)
+
+        ctx = MagicMock()
+        ctx.config.issue = 42
+        ctx.repo_root = "/fake/repo"
+
+        with patch("loom_tools.common.systematic_failure.record_blocked_reason"), \
+             patch("loom_tools.common.systematic_failure.detect_systematic_failure"):
+            _mark_judge_exhausted(ctx, 1)
+
+        # First subprocess call should be the label transition
+        label_call = mock_run.call_args_list[0]
+        cmd = label_call[0][0]
+        assert "gh" in cmd
+        assert "--remove-label" in cmd
+        assert "loom:building" in cmd
+        assert "--add-label" in cmd
+        assert "loom:blocked" in cmd
+
+    @patch("subprocess.run")
+    def test_records_blocked_reason(self, mock_run: MagicMock) -> None:
+        """Should record judge_exhausted as the blocked reason."""
+        mock_run.return_value = MagicMock(returncode=0)
+
+        ctx = MagicMock()
+        ctx.config.issue = 42
+        ctx.repo_root = "/fake/repo"
+
+        with patch("loom_tools.common.systematic_failure.record_blocked_reason") as mock_record, \
+             patch("loom_tools.common.systematic_failure.detect_systematic_failure"):
+            _mark_judge_exhausted(ctx, 2)
+
+        mock_record.assert_called_once_with(
+            "/fake/repo",
+            42,
+            error_class="judge_exhausted",
+            phase="judge",
+            details="judge failed after 2 retry attempt(s)",
+        )
+
+    @patch("subprocess.run")
+    def test_adds_diagnostic_comment(self, mock_run: MagicMock) -> None:
+        """Should add a comment with diagnostic info."""
+        mock_run.return_value = MagicMock(returncode=0)
+
+        ctx = MagicMock()
+        ctx.config.issue = 42
+        ctx.repo_root = "/fake/repo"
+
+        with patch("loom_tools.common.systematic_failure.record_blocked_reason"), \
+             patch("loom_tools.common.systematic_failure.detect_systematic_failure"):
+            _mark_judge_exhausted(ctx, 1)
+
+        # Second subprocess call should be the comment
+        comment_call = mock_run.call_args_list[1]
+        cmd = comment_call[0][0]
+        assert "comment" in cmd
+        body_idx = cmd.index("--body") + 1
+        assert "Shepherd blocked" in cmd[body_idx]
+        assert "1 retry" in cmd[body_idx]

--- a/loom-tools/tests/shepherd/test_config.py
+++ b/loom-tools/tests/shepherd/test_config.py
@@ -168,7 +168,19 @@ class TestShepherdConfig:
         """Default retry limits should be set."""
         config = ShepherdConfig(issue=42)
         assert config.doctor_max_retries == 3
+        assert config.judge_max_retries == 1
         assert config.stuck_max_retries == 1
+
+    def test_judge_max_retries_env_override(self) -> None:
+        """LOOM_JUDGE_MAX_RETRIES env var should override default."""
+        with patch.dict(os.environ, {"LOOM_JUDGE_MAX_RETRIES": "3"}):
+            config = ShepherdConfig(issue=42)
+            assert config.judge_max_retries == 3
+
+    def test_judge_max_retries_explicit(self) -> None:
+        """judge_max_retries can be explicitly set."""
+        config = ShepherdConfig(issue=42, judge_max_retries=5)
+        assert config.judge_max_retries == 5
 
     def test_worktree_marker_file(self) -> None:
         """Worktree marker file should have default value."""

